tests/test-generic-sort.cpp

```diff
@@ -70,5 +70,36 @@ int main()
   assert(us1 != us3);
   assert(us1->get_uninterpreted_name() == "sort1");
   assert(us1->get_arity() == 0);
+  cout << "pre datatype" << endl;
+  DatatypeDecl new_dt_decl = make_shared<GenericDatatypeDecl>("testSort1");
+  shared_ptr<GenericDatatype> new_dt =
+      shared_ptr<GenericDatatype>(new GenericDatatype(new_dt_decl));
+  shared_ptr<GenericDatatypeConstructorDecl> new_dt_cons_decl =
+      shared_ptr<GenericDatatypeConstructorDecl>(
+          new GenericDatatypeConstructorDecl("Cons"));
+  new_dt->add_constructor(new_dt_cons_decl);
+  shared_ptr<GenericDatatypeSort> dt_sort =
+      make_shared<GenericDatatypeSort>(new_dt);
+
+  DatatypeDecl new2_dt_decl = make_shared<GenericDatatypeDecl>("testSort2");
+  shared_ptr<GenericDatatype> new2_dt =
+      shared_ptr<GenericDatatype>(new GenericDatatype(new2_dt_decl));
+  shared_ptr<GenericDatatypeConstructorDecl> new2_dt_cons_decl =
+      shared_ptr<GenericDatatypeConstructorDecl>(
+          new GenericDatatypeConstructorDecl("test2"));
+  new2_dt->add_constructor(new2_dt_cons_decl);
+
+  shared_ptr<GenericDatatypeSort> dt_sort2 =
+      make_shared<GenericDatatypeSort>(new2_dt);
+  assert(dt_sort != dt_sort2);
+  auto copy = dt_sort;
+  assert(dt_sort == copy);
+  cout << dt_sort->compute_string() << endl;
+  cout << dt_sort2->compute_string() << endl;
+  assert(dt_sort->to_string() != dt_sort2->to_string());
+  assert((dt_sort->get_sort_kind()) == (dt_sort2->get_sort_kind()));
+  assert(dt_sort->get_sort_kind() == DATATYPE);
+  assert(dt_sort2->get_sort_kind() == DATATYPE);
```

Comment:
Thanks for adding this test! Can you please add documentation comments for it as well? 
Like "create the datatype" before the block that creates it, "compare equal datatypes" before some assertions etc.